### ISSUE TITLE
migrating from 2.0.12 to latest (3.1.2) : can't access class in the bound libwasm.js

### ISSUE THREAD

**Issue by flatmax** [emscripten-core/emsdk#1198]
Relating to the upgrade from v2 to v3 (https://github.com/emscripten-core/emsdk/issues/1194), this issue is to do with accessing a class which is bound in a emscripten output file.

In this [example WASM C++ autotools package](https://github.com/flatmax/WASM) a loadable WASM module is output and it is loaded in the browser [like so](https://github.com/flatmax/WASM/blob/master/webcomponent/libWASM.js#L2)  :
```
let modProm = import('./libwasm.js');
```

A [Test C++ class](https://github.com/flatmax/WASM/blob/master/include/Test.H#L34) is linked into the module, however this class can't be instantiated, for example the following code fails :

```
modProm.then((mod)=>{
      this.libWASM = mod; // for rendered wasm that delay
      let test = new this.libWASM.Test;
    })
```

**Comment by flatmax** [comment]
I have a branch now which outputs one js file for both nodejs and the brwoser. The nodejs version can find the Test class. The browser versions can't find it. https://github.com/flatmax/WASM/tree/oneJSFile

**Comment by flatmax** [comment]
I got this working for the browser, however it requires the following setup ...

When binding (this doesn't work with nodejs) :  -sEXPORT_ES6=1 

For instantiation :

```
import modProm from './libwasm.js';
modProm().then((mod)=>{
   this.libWASM = mod;
   let test = new this.libWASM.Test;
   test.sayHello();
})
```

If "-sEXPORT_ES6=1" is removed then instantiation doesn't work.

**Comment by flatmax** [comment]
Yes, ES6 module importing requires -SEXPORT_ES6. It doesn't work without that. It breaks the nodejs operation but fixes the browser operation. Here is the working bind command ([from here](https://github.com/flatmax/WASM/blob/oneJSFile/src/Makefile.am#L38)) :
```
@emcc -lembind -sSINGLE_FILE -sALLOW_MEMORY_GROWTH=1 -sEXPORT_ES6=1 -sMODULARIZE -sEXPORT_NAME="libwasm" $(AM_CXXFLAGS) .libs/libwasm.so -o .libs/libwasm.js
```

Removing EXPORT_ES6 allows the binary to run in nodejs but the module doesn't have the C++ classes exposed in the browser.

**Comment by sbc100** [comment]
How are you include the generated .js file in the your browser context?  How are you accessing the module in the browser context? 

I suggest you *don't* add `EXPORT_ES6` for now and instead look into fixing the non-es6-module use case in node.  That is very well tested so we should be able to figure out what is wrong in that case and get it working.

**Comment by flatmax** [comment]
When I remove EXPORT_ES6, I console.log the mod :

```
let modProm = import('./libwasm.js');
modProm.then((mod)=>{
  console.log(mod)
  let test = new mod.Test;
  test.sayHello();
})
```
There is nothing inside the module :
```
Module {Symbol(Symbol.toStringTag): 'Module'}
Symbol(Symbol.toStringTag) : "Module"
```

I am running v3.1.26.

**Comment by flatmax** [comment]
In nodejs, like so :
```
libWASM = require('./libwasmNode.js');
libWASM().then((mod)=>{
  console.log(mod)
  libWASM = mod;
  let test = new libWASM.Test;
  test.sayHello();
});
```

Everything is there :
```
<ref *1> {
  ready: Promise { [Circular *1] },
  inspect: [Function (anonymous)],
  HEAP8: Int8Array(16777216) [
    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0,
    ... 16777116 more items
  ],
  HEAP16: Int16Array(8388608) [
... SNIP ...
```

**Comment by sbc100** [comment]
Closing out this old issue.  If you are still having troubles like this please comment or re-open.
